src/components/sdfgViewer.ts

```diff
@@ -284,20 +341,38 @@ export class SdfgViewerProvider
         );
     }
 
-    public openViewer(uri: vscode.Uri, zoom_to: string | undefined = undefined,
-        display_bps: boolean = false) {
-        vscode.commands.executeCommand("vscode.openWith", uri, SdfgViewerProvider.viewType).then(_ => {
-            if (zoom_to)
-                this.handleMessage({
-                    type: 'zoom_to_node',
-                    uuid: zoom_to,
-                });
-            if (display_bps)
-                this.handleMessage({
-                    type: 'display_breakpoints',
-                    display: display_bps
+    public openViewer(uri: vscode.Uri, messages: Message[] = []) {
+        // If the SDFG is currently open, then execute the messages
+        // otherwise store them to execute as soon as the SDFG is loaded
+        let editorIsLoaded = false;
+        for (const editor of this.getOpenEditors()) {
+            if (editor.document.uri.fsPath === uri.fsPath) {
+                editorIsLoaded = true;
+                break;
+            }
+        }
+        if (!editorIsLoaded) {
+            for (const msg of messages) {
+                this.messages.push(msg);
+            }
+            vscode.commands.executeCommand("vscode.openWith", uri, SdfgViewerProvider.viewType);
+        }
+        else
+            vscode.commands.executeCommand("vscode.openWith", uri, SdfgViewerProvider.viewType).then(_ => {
+                messages.forEach(msg => {
+                    msg.sendMessage();
```

Comment:
As the webview is open but not focused it's already loaded so messages can be handled directly afterwards.